### PR TITLE
Fix compile error: bcc_elf.c: undefined reference to `debuginfod_begin'

### PR DESCRIPTION
**Pull request by Rtoax**
If libdebuginfod is installed on the system and bpftrace is compiled with static linking to BCC (STATIC_BPF_BCC), then libbcc.a requires the dynamic library -ldebuginfod, in which case a compilation error will be reported if bpftrace is compiled without linking libdebuginfod. We just need to add bcc's FindLibDebuginfod.cmake to bpftrace to solve this problem.

## How to reproduce the error:

```
$ sudo dnf install -y elfutils-debuginfod*
$ cd bpftrace
$ mkdir build && cd build
$ ../build-libs.sh
$ cmake -DCMAKE_INSTALL_PREFIX=/usr/ -DCMAKE_BUILD_TYPE=Debug .. $ make -j$(nproc)
...
[ 91%] Linking CXX executable bpftrace
[ 92%] Linking CXX executable bpftrace_test
/usr/bin/ld: ../build-libs/lib64/libbcc.a(bcc_elf.c.o): in function `find_debug_file': bcc_elf.c:(.text+0x10df): undefined reference to `debuginfod_begin' /usr/bin/ld: bcc_elf.c:(.text+0x10fd): undefined reference to `debuginfod_find_debuginfo' /usr/bin/ld: bcc_elf.c:(.text+0x1110): undefined reference to `debuginfod_end' collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/bpftrace.dir/build.make:121: src/bpftrace] Error 1 make[1]: *** [CMakeFiles/Makefile2:1328: src/CMakeFiles/bpftrace.dir/all] Error 2 make[1]: *** Waiting for unfinished jobs....
/usr/bin/ld: ../build-libs/lib64/libbcc.a(bcc_elf.c.o): in function `find_debug_file': bcc_elf.c:(.text+0x10df): undefined reference to `debuginfod_begin' /usr/bin/ld: bcc_elf.c:(.text+0x10fd): undefined reference to `debuginfod_find_debuginfo' /usr/bin/ld: bcc_elf.c:(.text+0x1110): undefined reference to `debuginfod_end' collect2: error: ld returned 1 exit status
make[2]: *** [tests/CMakeFiles/bpftrace_test.dir/build.make:364: tests/bpftrace_test] Error 1 make[1]: *** [CMakeFiles/Makefile2:2230: tests/CMakeFiles/bpftrace_test.dir/all] Error 2 make: *** [Makefile:146: all] Error 2
```

### CODE REVIEW

File: cmake/FindLibDebuginfod.cmake [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+
+# handle the QUIETLY and REQUIRED arguments and set LIBDEBUGINFOD_FOUND to TRUE if all listed variables are TRUE
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(LibDebuginfod DEFAULT_MSG
+  LIBDEBUGINFOD_LIBRARIES
+  LIBDEBUGINFOD_INCLUDE_DIRS)
+
+if (LIBDEBUGINFOD_FOUND)
+  add_definitions(-DHAVE_LIBDEBUGINFOD)
+endif (LIBDEBUGINFOD_FOUND)
```
Comment (danobi) [inline cmake/FindLibDebuginfod.cmake:53]: Unused? Any code actually using this macro?

**Comment by BurntBrunch** [comment]
I landed https://github.com/iovisor/bcc/pull/4308 in bcc to add ENABLE_LIBDEBUGINFOD, so we can configure it without it. We could use this as an opportunity to test the vendored libs flow:
1) update the bcc submodule to that commit
2) use the new flag in build-libs.sh
3) note somewhere (v0.17 issue?) that we need this commit to make it to an official bcc version before we push out v0.17 (though that's debatable!)

File: cmake/FindLibDebuginfod.cmake (hunk 1 above)
Comment (Rtoax) [reply cmake/FindLibDebuginfod.cmake:53]: FindLibDebuginfod.cmake totally copy from bcc/cmake, when bpftrace link to libbcc.a(libbcc.a link to libdebuginfod), if bpftrace not link to libdebuginfod, compile fatal when link.

**Comment by Rtoax** [comment]
@BurntBrunch thanks a lot, i'll try use ENABLE_LIBDEBUGINFOD.

**Comment by Rtoax** [comment]
@BurntBrunch @danobi Hi, i just start a new PR https://github.com/iovisor/bpftrace/pull/2415, maybe close this PR later.

**Pull request closed without merging**